plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -163,12 +163,14 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
 
         if (fileScanTaskIterable == null) {
             // Used to avoid duplicating work if the Dynamic Filter was already pushed down to the Iceberg API
+            boolean dynamicFilterIsComplete = dynamicFilter.isComplete();
```

Comment:
@raunaqmorarka can you sanity check for me that `isComplete()` means that the filter we get from `getCurrentPredicate()` will not change later in the query?

Reply:
yes, that is right